test/tests/meshgenerators/refine_block_generator/tests

```diff
@@ -0,0 +1,36 @@
+[Tests]
+  issues = '#18686'
+  design = 'meshgenerators/RefineBlockGenerator.md'
+
+  [generate]
+    requirement = 'The system shall support the ability to refine blocks'
+
+    [single_block]
+      type = 'Exodiff'
+      input = 'test_single.i'
+      exodiff = 'test_single_in.e'
+      cli_args = '--mesh-only'
+      recover = false
+      detail = 'in a mesh with a single first order element'
+
+    []
+    [multiple_block]
+      type = 'Exodiff'
+      input = 'test_multi.i'
+      exodiff = 'test_multi_in.e'
+      recover = false
+      cli_args = "--mesh-only"
+
+      detail = 'in a Cartesian 3D mesh.'
+    []
+    [neighbor_refinement]
+      type = 'Exodiff'
+      input = 'test_neighbor.i'
+      exodiff = 'test_neighbor_in.e'
+      cli_args = "--mesh-only"
+      recover = false
+      detail = 'with the neighbor refinement on'
```

Comment:
describe more. 'the neighbor refinement on' feels like what we chose as a parameter. SQA is more about what the code does than what we named things